card/mod/carrierwave/set/all/file_utils.rb

```diff
@@ -5,11 +5,20 @@ def update_all_storage_locations
   end
 
   def delete_tmp_files_of_cached_uploads
+    cards_with_disposable_attachments do |card, action|
+      card.delete_files_for_action action
+      action.delete
+    end
+  end
+
+  def cards_with_disposable_attachments
     draft_actions_with_attachment.each do |action|
       # we don't want to delete uploads in progress
       next unless old_enough?(action.created_at) && (card = action.card)
-      card.delete_files_for_action action
-      action.delete
+      # we can't delete attachments we don't have write access to
+      next if card.read_only?
```

Comment:
somewhat surprised that code climate complains about cognitive complexity here, but it would be dead easy to grok if it were:

```
draft_actions_with_attachment.each do |action|
  next unless disposable_action? action
  yield action
end
```

Reply:
actually, you can just leave it at `old_enough?` if the above were:
```
disposable_attachment_actions do |action|
  next if action.card&.read_only?
  action.delete
end
```
This would change the logic slightly and delete actions without cards, but I'd consider that a good thing :)